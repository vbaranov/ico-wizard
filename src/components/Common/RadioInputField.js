import React from 'react';
import '../../assets/stylesheets/application.css';

export const RadioInputField = (props) => {
  const inputs = props.items
    .map((item, index) => (
      <label className="radio-inline" key={index}>
        <input
          type="radio"
          checked={props.selectedItem === item.value}
          onChange={props.onChange}
          value={item.value}
        />
        <span className="title">{item.label}</span>
      </label>
    ))

  return (
    <div className={props.extraClassName}>
      <label className="label">{props.title}</label>
      <div className="radios-inline">
        { inputs }
      </div>
      <p className="description">{props.description}</p>
    </div>
  );
}
