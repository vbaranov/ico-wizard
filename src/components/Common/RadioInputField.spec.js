import React from 'react';
import { RadioInputField } from './RadioInputField';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import { configure, mount } from 'enzyme';

configure({ adapter: new Adapter() })

test('Should render the component', () => {
  const component = renderer.create(
    <RadioInputField
      title="Some Title"
      items={[{ label: 'Item 1', value: 'item-1' }, { label: 'Item 2', value: 'item-2' }]}
      selectedItem="item-1"
      onChange={() => {}}
      description="Some Description"
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should render two inputs and check the first one', () => {
  const wrapper = mount(
    <RadioInputField
      title="Some Title"
      items={[{ label: 'Item 1', value: 'item-1' }, { label: 'Item 2', value: 'item-2' }]}
      selectedItem="item-1"
      onChange={() => {}}
      description="Some Description"
    />
  );

  const inputs = wrapper.find('input[type="radio"]')
  expect(inputs.length).toBe(2)

  const firstInput = inputs.at(0)
  const secondInput = inputs.at(1)

  expect(firstInput.prop('checked')).toBe(true)
  expect(secondInput.prop('checked')).toBe(false)
})

test('The onChange callback is called when an item is selected', () => {
  const onChangeCallback = jest.fn()

  const wrapper = mount(
    <RadioInputField
      title="Some Title"
      items={[{ label: 'Item 1', value: 'item-1' }, { label: 'Item 2', value: 'item-2' }]}
      selectedItem="item-1"
      onChange={onChangeCallback}
      description="Some Description"
    />
  );

  wrapper.find('input[type="radio"]').at(1).simulate('change')

  expect(onChangeCallback).toBeCalled()
})

test('Should change the selected item when `selectedItem` changes', () => {
  const onChangeCallback = jest.fn()

  const wrapper = mount(
    <RadioInputField
      title="Some Title"
      items={[{ label: 'Item 1', value: 'item-1' }, { label: 'Item 2', value: 'item-2' }]}
      selectedItem="item-1"
      onChange={onChangeCallback}
      description="Some Description"
    />
  );

  let inputs = wrapper.find('input[type="radio"]')
  expect(inputs.length).toBe(2)

  let firstInput = inputs.at(0)
  let secondInput = inputs.at(1)

  expect(firstInput.prop('checked')).toBe(true)
  expect(secondInput.prop('checked')).toBe(false)

  wrapper.setProps({ selectedItem: 'item-2' })

  inputs = wrapper.find('input[type="radio"]')
  expect(inputs.length).toBe(2)

  firstInput = inputs.at(0)
  secondInput = inputs.at(1)

  expect(firstInput.prop('checked')).toBe(false)
  expect(secondInput.prop('checked')).toBe(true)
})
