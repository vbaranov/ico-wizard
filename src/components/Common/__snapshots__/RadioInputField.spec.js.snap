// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the component 1`] = `
<div
  className={undefined}
>
  <label
    className="label"
  >
    Some Title
  </label>
  <div
    className="radios-inline"
  >
    <label
      className="radio-inline"
    >
      <input
        checked={true}
        onChange={[Function]}
        type="radio"
        value="item-1"
      />
      <span
        className="title"
      >
        Item 1
      </span>
    </label>
    <label
      className="radio-inline"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="radio"
        value="item-2"
      />
      <span
        className="title"
      >
        Item 2
      </span>
    </label>
  </div>
  <p
    className="description"
  >
    Some Description
  </p>
</div>
`;
